tests/validation/test_ou3_p4_thirdgen_combined_certificate.py

```
@@ -0,0 +1,28 @@
+ROOT=Path(__file__).resolve().parents[2]
+sys.path.insert(0,str(ROOT/'tools'))
+import ou3_p4_thirdgen_combined_certificate as C
+import ou3_validate_enclosure as ENC
+class ThirdGenCombinedP4Tests(unittest.TestCase):
+ @classmethod
+ def setUpClass(cls): cls.d=C.build()
+ def test_combined_passes(self):
+  self.assertEqual(C.validate(self.d),[]); self.assertEqual(self.d['P4_THIRDGEN_COMBINED_WORD_CERTIFICATE'],'PASS'); self.assertTrue(self.d['thirdgen_stacks_431_and_432_refinements'])
+ def test_combined_never_regresses_431(self):
+  for mode in ('H','A'):
+   m=self.d['modes'][mode]; self.assertGreaterEqual(m['certified_level_W'],m['certified_level_W_before_combined']); self.assertGreaterEqual(m['thirdgen_W_widening_factor_vs_431_lower'],1.0); self.assertGreater(m['thirdgen_total_W_widening_factor_vs_legacy_lower'],1.0)
```

Ruff (0.16.2)

[error] Multiple statements on one line (semicolon)

(E702)